backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -315,16 +285,110 @@ public CreateCommitRequest.Response setCommitFromDatasetVersion(
     }
   }
 
+  private Blob getBlobBuilderFromDatasetVersion(DatasetVersion datasetVersion) {
+    var blobBuilder = Blob.newBuilder();
+    if (datasetVersion.hasDatasetBlob()) {
+      blobBuilder.setDataset(populateInternalPathInDatasetBlob(datasetVersion.getDatasetBlob()));
+      return blobBuilder.build();
+    }
```

Comment:
since this block is now nice and small, I probably wouldn't re-use the `blobBuilder` variable at all, but just create one and return it immediately inside this block. Then, you can move the `blobBuilder`  usage down inside the block below and not even keep it around as a variable, for simplicity.